### DISCUSSION TITLE
html2canvas failing when trying to capture embedded SVG

### DISCUSSION THREAD

**Discussion by redyczek** [niklasvh/html2canvas#2855] [Q&A]
I'm using html2canvas to capture a diagram (bar chart), created by SVGGraph and rendered as an SVG. The chart is inside a `<div>` which is the target of html2canvas. The `<div>`s have a yellow background to better see what happens.

If I include the SVG using an `<img>` tag, html2canvas works like a charm. However, that way I lose the interactivity of the diagram (tooltips, links, etc.)

If I include the SVG using either an `<embed>` tag or an `<iframe>`, I get the interactivity, but html2canvas fails (empty image). 

All files (html, php, js) are on the same server.

Is there a known limitation / issue with capturing (interactive) SVGs, or this maybe an indication of a problem with the SVG?

I've set up a simple public test page here: http://gobo-net.de/h2c/
- Copying the graph in the <img> tag works (appended at end of document)
- Copying the graph in the <embed> tag doesn't work (adding empty yellow space at end of document)

-----------------------------------------------------------------------------------------------------------------

_(The following part of the original comment was identified as having nothing to do with the issues)_

_Google Chrome does give me the following error message (when using `<embed>` tag or an `<iframe>`) , but I'm not sure whether that's even related to html2canvas:_

```
Uncaught TypeError: Cannot read properties of null (reading 'appendChild')
    at XMLDocument.<anonymous> (chrome-extension://bpjfchmapbmjeklgmlkabfepflgfckip/src/content/sidebar.js:812:17)
```

**Comment by coder0107git** [comment]
When I lookup the chrome extension by the id I find [Password Safe by MATESO](https://chrome.google.com/webstore/detail/password-safe-by-mateso/bpjfchmapbmjeklgmlkabfepflgfckip). When I look through the source code of the src/content/sidebar.js I can't find any problems. Even though I don't have that extension installed I get the same results.

**Reply by redyczek** [reply]
Thanks for the reply! I've updated my original post accordingly.

**Reply by ionansantos** [reply]
up

**Comment by ionansantos** [comment]
I have a similar problem, I can't capture an iframe from spotify, youtube, the capture is empty does anyone know how to solve it?

**Reply by coder0107git** [reply]
Sorry that this is so late. I think you can't capture an iframe from a different origin without running into CORS problems unless you use a CORS proxy. If you aren't already using some kind of CORS proxy I would suggest it because html2canvas by itself does not bypass CORS restrictions. In the [html2canvas documentation](https://html2canvas.hertzen.com/documentation) it suggest using the [html2canvas-proxy](https://github.com/niklasvh/html2canvas-proxy-nodejs). For more information about it see [here](https://html2canvas.hertzen.com/proxy).
